src/frontend/src/handler/create_source.rs

```diff
@@ -1188,7 +1190,57 @@ pub async fn check_iceberg_source(
         .collect::<Vec<_>>();
     let new_iceberg_schema = arrow_schema::Schema::new(new_iceberg_field);
 
-    risingwave_connector::sink::iceberg::try_matches_arrow_schema(&schema, &new_iceberg_schema)?;
+    try_matches_arrow_schema_for_iceberg_source(&schema, &new_iceberg_schema)?;
+
+    Ok(())
+}
+
+/// Try to match our schema with iceberg schema.
+pub fn try_matches_arrow_schema_for_iceberg_source(
```

Comment:
Because we need to support reading decimal type data from the Iceberg source，i.e. Iceberg decimal -> risingwave decimal. However, iceberg sink will convert decimal type to large binary type which is unexpected for iceberg source.

Reply:
That's a simplified workaround, and we will eventually merge them. How about putting them in same method and add some arguments(e.g. `is_source: bool`) so that we will not diverge too much?

Reply:
Sure